simple/poll.c

```diff
@@ -105,22 +118,26 @@ static int send_recv()
 	struct fi_cq_entry comp;
 	int ret;
 	int ret_count = 0;
-	int i;
-
-//	fprintf(stdout, "Posting a recv...\n");
-//	ret = ft_post_rx(rx_size);
-//	if (ret)
-//		return ret;
+	int i, tx_cntr_done = 0, rx_cntr_done = 0;
 
 	fprintf(stdout, "Posting a send...\n");
 	ret = ft_post_tx(ep, remote_fi_addr, tx_size, &tx_ctx);
 	if (ret)
 		return ret;
 
-	while ((tx_cq_cntr < tx_seq) || (rx_cq_cntr < rx_seq)) {
+	/* wait for TX/RX counter events from init-AV */
+	ret = fi_cntr_wait(txcntr, 1, 2000);
+	if (ret)
+		return ret;
+	ret = fi_cntr_wait(rxcntr, 1, 2000);
+	if (ret)
+		return ret;
```

Comment:
ft_tx/rx in ft_init_av calls ft_get_tx/rx_comp which would call fi_cntr_wait. So we need not call it again here.